### PR TITLE
Ability to disable entrypoint and /api/docs

### PR DESCRIPTION
**Pull request by athos7933**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #1568
| License       | MIT
| Doc PR        | api-platform/doc#365

### CODE REVIEW

File: src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -87,6 +87,7 @@ public function getConfigTreeBuilder()
                 ->end()
                 ->booleanNode('enable_swagger')->defaultValue(true)->info('Enable the Swagger documentation and export.')->end()
                 ->booleanNode('enable_swagger_ui')->defaultValue(class_exists(TwigBundle::class))->info('Enable Swagger ui.')->end()
+                ->booleanNode('enable_entrypoint')->defaultTrue()->info('Enable/disable the entry-point')->end()
```
Comment (dunglas) [inline src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php]: `Enable the entrypoint` for the shake of consistency

File: src/Bridge/Symfony/Bundle/Resources/config/routing/doc.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (dunglas) [inline src/Bridge/Symfony/Bundle/Resources/config/routing/doc.xml]: Should `docs.xml`.

File: src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php (hunk 1 above)
Comment (dunglas) [reply src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php]: I would also create a separate flag for entrypoint and docs, (you may want the docs, but not the entrypoint for instance).

**Comment by athos7933** [comment]
I dont't know why some behat failed ! My changes do not impact the tests.

**Comment by soyuka** [comment]
We need to merge 2.1, it's because of #1578

**Comment by Nek-** [comment]
@athos7933 you need to rebase your work on master :)

**Comment by soyuka** [comment]
Just merged 2.1, please rebase :)

**Comment by athos7933** [comment]
@soyuka Yes, i'ts already done

**Comment by Nek-** [comment]
@athos7933 I don't know what you did wrong but what you did is not a rebase but a merge.

File: src/Bridge/Symfony/Routing/ApiLoader.php [hunk 3]
```diff
@@ -146,6 +146,14 @@ private function loadExternalFiles(RouteCollection $routeCollection)
     {
         $routeCollection->addCollection($this->fileLoader->load('api.xml'));
 
+        if ($this->container->getParameter('api_platform.enable_entrypoint')) {
```
Comment (meyerbaptiste) [inline src/Bridge/Symfony/Routing/ApiLoader.php]: Should not we inject these parameters in the constructor instead?

Comment (Nek-) [reply src/Bridge/Symfony/Routing/ApiLoader.php]: The constructor is already really huge. Also the container is already use at many places. This seems ok.

**Comment by athos7933** [comment]
@dunglas I did the change you want.

File: src/Bridge/Symfony/Routing/ApiLoader.php (hunk 3 above)
Comment (greg0ire) [reply src/Bridge/Symfony/Routing/ApiLoader.php]: Another option is to use… options, along with the option resolver

Comment (Nek-) [reply src/Bridge/Symfony/Routing/ApiLoader.php]: @greg0ire it's adding a dependency. Which is not good.

**Comment by Nek-** [comment]
Failure is not related to this patch, it's about Symfony4 and this new line of code: https://github.com/symfony/symfony/blame/master/src/Symfony/Component/HttpFoundation/Request.php#L1855

File: src/Bridge/Symfony/Routing/ApiLoader.php (hunk 3 above)
Comment (greg0ire) [reply src/Bridge/Symfony/Routing/ApiLoader.php]: You can use a simple array, that's possible too

**Comment by vincentchalamon** [comment]
Hello @Nek- @athos7933, unit tests will be fixed in this PR: https://github.com/api-platform/core/pull/1631

**Comment by athos7933** [comment]
@dunglas Hi ! Waiting for review.

**Comment by dunglas** [comment]
Thanks you very much @athos7933! Don't forget to open a docs PR against the master branch please :)

File: src/Bridge/Symfony/Routing/ApiLoader.php [hunk 4]
```diff
@@ -146,6 +150,14 @@ private function loadExternalFiles(RouteCollection $routeCollection)
     {
         $routeCollection->addCollection($this->fileLoader->load('api.xml'));
 
+        if ($this->entrypointEnabled) {
+            $routeCollection->addCollection($this->fileLoader->load('api.xml'));
+        }
```
Comment (Taluu) [inline src/Bridge/Symfony/Routing/ApiLoader.php]: doesn't this add 2 times the `api.xml` file ? :o

Comment (Taluu) [reply src/Bridge/Symfony/Routing/ApiLoader.php]: (looking at the line above)

Comment (soyuka) [reply src/Bridge/Symfony/Routing/ApiLoader.php]: thanks

**Pull request merged**